src/services/apiUtils.ts

```diff
@@ -0,0 +1,40 @@
+import { getUniqueId } from '@patternfly/react-core';
+import { QueryClient } from '@tanstack/react-query';
+
+interface ApiHandlerProps<T> {
+  apiServiceMethod: Function;
+  url: string;
+  payload: T;
+  successMessage: string;
+  errorMessage: string | null;
```

Comment:
would rather see undefined than null, which can also be done like
```suggestion
  errorMessage?: string;
```